Mage/src/main/java/mage/ConditionalMana.java

```diff
@@ -220,4 +227,42 @@ public void add(ManaType manaType, int amount) {
                 break;
         }
     }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(super.hashCode(), conditions, staticText, scope, manaProducerId, manaProducerOriginalId);
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Must override `hashCode` and `equals` for inherited `ConditionalMana` classes with extra fields. See example:
![shot_220712_103027](https://user-images.githubusercontent.com/8344157/178430502-148429b3-9a6b-4adf-95c5-eaa271537b75.png)

`ConditionalMana` must have docs about override requirements (see above).
![shot_220712_111210](https://user-images.githubusercontent.com/8344157/178430553-6df256b9-8dcc-4bbb-87be-5aef181c1a4a.png)

Reply:
Well, I checked all inherited `ConditionalMana`, no inner fields. So it's fine. But must contains docs about it anyway.

Reply:
I will go through and check them. Also, I feel like ConditionalManaBuilder needs a better constructor. There are dozens of implementations of it which implement the exact same thing (a build with a single instance of a ConditionalMana object and a rule string)

Reply:
The equals and hashCode I will handle most of the rest inside #9225.